crates/transcode/src/lib.rs

```diff
@@ -347,6 +353,24 @@ impl ContractMessageTranscoder {
             Ok(Value::Unit)
         }
     }
+
+    /// Checks if buffer empty, otherwise returns am error
+    fn validate_length(data: &[u8], label: &str, args: &[(Value, Value)]) -> Result<()> {
```

Comment:
Extracted check to a separate function instead of moving it to a higher layer because each decode function returns a generic `anyhow` error. We would need to return a custom error and map all `anyhow` errors inside each `decode` function to a specific error. We will also need to handle the error on the higher `decode` function. Additionally, events and messages/constructors have different structs that are used to extract `label`. That will also unnecessary overcomplicate error handling.

Therefore, it is better that all validation happens on the lower level at each `decode` function independently.